server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructConfigurationServices.test.js

```diff
@@ -0,0 +1,376 @@
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => 'ro-2-0-1-op-s-bm-001');
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => {
+      return Promise.reject(mockError);
+    });
+
+    const res = ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    return res.then()
+    .catch((err)=> {
+      expect(err).toEqual(mockError);
+    })
+    
+  });
+
+  test("configureForwardingConstructAsync - ForwardingConstructForTheForwardingNameAsync returns undefined", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    const errorMsg = "Cannot read properties of undefined (reading 'fc-port')";
+    
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => 'ro-2-0-1-op-s-bm-001');
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => undefined);
+
+    const res = ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    return res.then()
+    .catch((err)=> {
+      expect(err.message).toStrictEqual(errorMsg);
+    })
+    
+  });
+
+  test("configureForwardingConstructAsync - failed to set LogicalTerminationPointAsync", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    
+    const mockError = { message: 'Something bad happened' };
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => 'ro-2-0-1-op-s-bm-001');
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fc);
+    jest.spyOn(fcPort, 'setLogicalTerminationPointAsync').mockImplementation(() => {
+      return Promise.reject(mockError);
+    });
+
+    const res = ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    return res.then()
+    .catch((err)=> {
+      expect(err).toEqual(mockError);
+    })
+    
+  });
+
+  test("configureForwardingConstructAsync - failed to configure ForwardingConstructAsync", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    
+    const mockError = { message: 'Something bad happened' };
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => 'ro-2-0-1-op-s-bm-001');
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fc);
+    jest.spyOn(fcPort, 'setLogicalTerminationPointAsync').mockImplementation(() => true);
+
+    const res = ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    return res.then(() => {
+      return Promise.reject(mockError);
```

Comment:
Here the '.then()' part is not only useless but misleading, because you are throwing the error you are expecting, which means the test will ALWAYS pass, no matter what 'res' contains.